gh-121249: Support _Complex types in the struct module (#121613)
python/cpython: Doc/library/struct.rst

| ``P``  | :c:expr:`void \*`        | integer            |                | \(5)       |
+--------+--------------------------+--------------------+----------------+------------+

> Additionally, if IEC 60559 compatible complex arithmetic (Annex G) is

Reviewer: This should probably link to the specification.

Author: Which link do you suggest?  Wikipedia page for IEEE 754, C11 standard?

Reviewer: Probably the C11 standard.

> Additionally, if IEC 60559 compatible complex arithmetic (Annex G) is
> supported, the following format characters are available:

Author: ```suggestion
Additionally, if IEC 60559 compatible complex arithmetic (Annex G of the
C11 standard) is supported, the following format characters are available:
```
?

Reviewer: That would be perfect.

+--------+--------------------------+--------------------+----------------+------------+
| Format | C Type                   | Python type        | Standard size  | Notes      |
+========+==========================+====================+================+============+
| ``E``  | :c:expr:`float complex`  | complex            | 8              | \(10)      |
+--------+--------------------------+--------------------+----------------+------------+
| ``C``  | :c:expr:`double complex` | complex            | 16             | \(10)      |
+--------+--------------------------+--------------------+----------------+------------+

.. versionchanged:: 3.3
   Added support for the ``'n'`` and ``'N'`` formats.

.. versionchanged:: 3.6
   Added support for the ``'e'`` format.

.. versionchanged:: 3.14

>    Added support for ``'E'`` and ``'C'`` formats.

Reviewer 2: ```suggestion
   Added support for the ``'E'`` and ``'C'`` formats.
```

To be consistent with the previous 'versionchanged' entries.

   of bytes.  As a special case, ``'0s'`` means a single, empty string (while
   ``'0c'`` means 0 characters).

(10)

>    For the ``'E'`` and ``'C'`` conversion codes, the packed representation uses

Reviewer 2: ```suggestion
   For the ``'E'`` and ``'C'`` format characters, the packed representation uses
```

I would use the same terminology as in the other notes.

>    the IEEE 754 binary32 and binary64 format for components of the complex

Reviewer: Again, link to the spec, we shouldn't expect people to have arbitrary specification numbers memorized.

Author: I doubt it's arbitrary.  We have many references to the IEEE 754 standard in docs.
